### PR TITLE
[ci][test-state/2] move test to different states

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
**Context:** This is a series of PR to compute test state (JAILED|PASSING|FAILING) on-the-fly, persist them into DB and create issues and bisect automatically, etc.

- Add a state machine to move test to different state under some conditions
- Actually move test to failing state, persist to DB

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests

### CODE REVIEW

**Comment by ollie-iterators** [comment]
The lint error has been fixed elsewhere

**Comment by aslonnie** [comment]
> The lint error has been fixed elsewhere. The dockerfile that the error is from is updated to ignore the error.

hi @ollie-iterators could you stop commenting on PR's when it is still in draft mode? thanks.  I think you often provide good info, but can be more helpful if you can understand when to speak and when not to. :)

**Comment by ollie-iterators** [comment]
> > The lint error has been fixed elsewhere. The dockerfile that the error is from is updated to ignore the error.
> 
> hi @ollie-iterators could you stop commenting on PR's when it is still in draft mode? thanks. I think you often provide good info, but can be more helpful if you can understand when to speak and when not to. :)

Ok

File: release/ray_release/test_automation/state_machine.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from ray_release.test import (
+    Test,
+    TestState,
+)
+
+
+class TestStateMachine:
```
Comment (aslonnie) [inline release/ray_release/test_automation/state_machine.py:7]: could you add a comment of what this state machine does at the high level?

like what is the end-goal logic that this is trying to implement?

**Review by aslonnie — APPROVED** [review]

**Comment by ollie-iterators** [comment]
The test_scheduling_2 tests can be fixed by merging in the recent changes from the main branch.

**Pull request merged**